hivemind/client/averaging/__init__.py

```diff
@@ -240,15 +246,26 @@ def run_in_background(self, await_ready=True, timeout=None):
 
     def shutdown(self) -> None:
         """ Shut down the averager process """
-        # TODO notify peers before terminating
-        if self._parent_pid != os.getpid() or self.is_alive():
-            self._pipe.send(('_SHUTDOWN', None))
-            self.terminate()
+        if self.is_alive():
+            self.pipe.send(('_shutdown', [None], {}))  # shut down the daemon process
```

Comment:
nit: I'd suggest to rename `self.pipe` and `self._pipe` to make clear the difference between them. If it is outside of scope of this PR, please create an issue.

Reply:
willdo